In [1]:
import pandas as pd
from scipy import stats
import statsmodels.formula.api as smf
import statsmodels.api as sm

In [13]:
df = pd.read_csv("/content/teachers_rating_data (3).csv")
df.head()

prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof38    Male   33    Yes        4.87        59     4.8       Lower
1  Prof38    Male   43    Yes        4.39        97     2.8       Lower
2  Prof46  Female   29     No        3.96        96     3.2       Lower
3  Prof46    Male   43     No        4.55        43     3.8       Lower
4  Prof13  Female   43    Yes        4.94        96     3.1       Lower

# Q1. Regression with T-test: Using the teachers rating data set, does gender affect teaching evaluation rates?

In [14]:
model = smf.ols('Evaluation ~ Gender', data=df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.033
Model:                            OLS   Adj. R-squared:                  0.025
Method:                 Least Squares   F-statistic:                     4.021
Date:                Mon, 27 Oct 2025   Prob (F-statistic):             0.0472
Time:                        13:52:21   Log-Likelihood:                -110.19
No. Observations:                 120   AIC:                             224.4
Df Residuals:                     118   BIC:                             230.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          3.9363      0.074     53.

Based on the regression results (F(1,118) = 4.02, p = 0.047), there is a statistically significant difference in teaching evaluation ratings between male and female instructors.

Male instructors scored about 0.23 points higher on average than female instructors (β = 0.226, p = 0.047). However, since the model explains only about 3.3% of the variance in evaluation scores (R² = 0.033), the practical impact of gender on teaching evaluations is quite small, even though the difference is statistically significant.

# Q2. Regression with ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

In [16]:
df['age_group'] = pd.cut(df['Age'], bins=[20, 30, 40, 50, 60, 70],
                         labels=['20s', '30s', '40s', '50s', '60s'])
model = smf.ols('Beauty ~ C(age_group)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)


                 sum_sq     df         F    PR(>F)
C(age_group)   2.394642    4.0  1.311694  0.273997
Residual      52.942685  116.0       NaN       NaN


/usr/local/lib/python3.12/dist-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 4, but rank is 3
  warnings.warn('covariance of constraints does not have full '


An ANOVA test was conducted to determine whether beauty scores differ by age group.
The results were not statistically significant, F(4,116) = 1.31, p = 0.274.
This indicates that instructors’ beauty scores do not significantly differ across age groups, suggesting that age group is not a major factor influencing beauty ratings in this sample.

# Q3. Correlation: Using the teachers' rating dataset, Is teaching evaluation score correlated with beauty score?

In [12]:
import statsmodels.api as sm

X = df['Beauty']
y = df['Evaluation']

X = sm.add_constant(X)

model = sm.OLS(y, X).fit()
predictions = model.predict(X)

print(model.summary())


                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.009
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     1.107
Date:                Mon, 27 Oct 2025   Prob (F-statistic):              0.295
Time:                        13:44:58   Log-Likelihood:                -97.548
No. Observations:                 120   AIC:                             199.1
Df Residuals:                     118   BIC:                             204.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.2624      0.263     16.228      0.0

An OLS regression was conducted to examine whether beauty scores predict teaching evaluations.
The model was not statistically significant, F(1,118) = 1.11, p = 0.295, with an R² = 0.009.
This indicates that beauty scores explain less than 1% of the variance in teaching evaluations. The coefficient for beauty (β = -0.071, p = 0.295) was not significant, suggesting that instructors’ beauty scores do not have a meaningful impact on their evaluation ratings in this dataset.